nodes/modifier_make/contour2D.py

```diff
@@ -0,0 +1,599 @@
+                    for i in range(vert):
+                        angL = lastAng + difAng / vert * i
+                        listVertX.append(x+Radius*cos(angL))
+                        listVertY.append(y+Radius*sin(angL))
+                        listVertZ.append(z)
+
+                lastIndex = ang[2]
+                lastAng = ang[1]
+                lastAngSide = ang[3]
+                lastAngTan = ang[5]
+                listVertX.append(x + Radius * cos(ang[0]))
+                listVertY.append(y + Radius * sin(ang[0]))
+                listVertZ.append(z)
+        elif connex == 1:
+            beta = (net[1] - net[2])
+            if beta < 0:
+                beta = (net[1] - net[2] + 2*pi)
+            vAngle += net[0] - net[1] - pi
+            vert = int((degrees(beta)) / theta) + 1
+            for i in range(vert):
+                localAngle = (radians(theta*i) + vAngle + 4*pi) % (2*pi)
+                listVertX.append(x + Radius * cos(localAngle))
+                listVertY.append(y + Radius * sin(localAngle))
+                listVertZ.append(z)
+
+            listVertX.append(x + Radius * cos(net[1]))
+            listVertY.append(y + Radius * sin(net[1]))
+            listVertZ.append(z)
+
+        else:
+            for i in range(vert):
+                localAngle = (radians(theta*i) + vAngle + 4*pi) % (2*pi)
+
+                listVertX.append(x + Radius * cos(localAngle))
+                listVertY.append(y + Radius * sin(localAngle))
+                listVertZ.append(z)
+        points = list((x, y, z) for x, y, z in zip(listVertX, listVertY, listVertZ))
+        return points
+
+    def sideEdges(self, v, edges, radius, net):
+        Verts = []
+        Edges = []
+        n = 0
+        netOffset = [0 for i in range(len(net))]
+
+        for ed in edges:
+            for s in ed:
+                netOf = netOffset[s]
+                netOffset[s] += 3
+                x1 = v[s][0] + radius[s] * cos(net[s][1 + netOf])
+                y1 = v[s][1] + radius[s] * sin(net[s][1 + netOf])
+                x2 = v[s][0] + radius[s] * cos(net[s][2 + netOf])
+                y2 = v[s][1] + radius[s] * sin(net[s][2 + netOf])
+
+                Verts.append((x1, y1, v[s][2]))
+                Verts.append((x2, y2, v[s][2]))
+
+            Edges.append((n, n+3))
+            Edges.append((n+1, n+2))
+            n += 4
+        return Verts, Edges
+
+    def make_edges(self, Vertices, net, pointsL):
+        listEdg = [(i, i + 1) for i in range(pointsL - 1)]
+        if len(net) == 0:
+            listEdg.append((pointsL-1, 0))
+        return listEdg
+
+    def make_faces(self, Vertices):
+        listPlg = list(range(Vertices))
+
+        return [listPlg]
+
+    def process(self):
+        inputs, outputs = self.inputs, self.outputs
+
+        # inputs
+
+        if inputs['Verts_in'].is_linked:
+
+            VertsAll = inputs['Verts_in'].sv_get(deepcopy=False)
+        else:
+            VertsAll = [[(0.0, 0.0, 0.0)]]
```

Comment:
consider using the `default` parameter, which will be used in case `.is_linked` returns False
```diff
-        if inputs['Verts_in'].is_linked:
-
-            VertsAll = inputs['Verts_in'].sv_get(deepcopy=False)
-        else:
-            VertsAll = [[(0.0, 0.0, 0.0)]]

+       VertsAll = inputs['Verts_in'].sv_get(deepcopy=False, default=[[(0.0, 0.0, 0.0)]])
or (not sure about the extra wrapping)
+       VertsAll = inputs['Verts_in'].sv_get(deepcopy=False, default=[(0.0, 0.0, 0.0)])
```